### DISCUSSION TITLE
The page component must have a unique name across all routes

### DISCUSSION THREAD

**Discussion by zanami** [blitz-js/blitz#2986] [General] [answered]
Please explain why? NextJS has no such requirement, AFAIK
See for example [here](https://github.com/vercel/commerce/blob/582e9257d97582564fded5463e35b1389e4aa945/pages/search/%5Bcategory%5D.tsx) and [here](https://github.com/vercel/commerce/blob/582e9257d97582564fded5463e35b1389e4aa945/pages/search/designers/%5Bname%5D/%5Bcategory%5D.tsx)

I'm trying to make a similar routing scheme as Vercel Commerce has

search.tsx
search/[category].tsx
search/designers/[name].tsx
etc.

They are reusing the same code in every page like this

```
import { getSearchStaticProps } from '@lib/search-props'
import type { GetStaticPathsResult, GetStaticPropsContext } from 'next'
import Search from '@components/search'

export async function getStaticProps(context: GetStaticPropsContext) {
  return getSearchStaticProps(context)
}

export function getStaticPaths(): GetStaticPathsResult {
  return {
    paths: [],
    fallback: 'blocking',
  }
}

export default Search
```

I guess unique names are required just for the [Route Manifest](https://blitzjs.com/docs/route-manifest) to make sense, right?

Is there a better way to do the same, like [[catchall]] routes maybe?

**Comment by beerose** [comment, marked as answer]
> I guess unique names are required just for the Route Manifest to make sense, right?

Yes, that's right. 

> Is there a better way to do the same, like [[catchall]] routes maybe?

Yes, you could catch all. In you case, you could have `app/pages/search/[[...slug]].tsx` and that will match /search, /search/[category], /search/designers/[name], and so on.

More here: https://blitzjs.com/docs/routing#optional-catch-all-routes
